scripts/loadAssets.py

```diff
@@ -0,0 +1,103 @@
+# File: loadAssets.py
+#
+# Author: Craig Cogdill
+
+import logging
+import logging.handlers
+import json
+import time
+from elasticsearch import Elasticsearch
+from os import listdir
+from os.path import isfile, join, splitext
+log_filename = "/var/log/probe/pythonKibanaStartup.log"
+logging.basicConfig(filename=log_filename,
+                    level=logging.DEBUG,
+                    format='%(asctime)s %(levelname)s:  %(message)s',
+                    datefmt='%Y/%m/%d %H:%M:%S')
+rotating_handler = logging.handlers.RotatingFileHandler(log_filename, 
+                                                        maxBytes=10485760,
+                                                        backupCount=5)
+es = Elasticsearch(timeout=5, max_retries=5, retry_on_timeout=True)
+
+kibana_version = "4.1.4"
+
+resources = "/usr/local/kibana-" + kibana_version + "-linux-x64/resources"
+dashboards_path = resources + "/dashboards" 
+visualizations_path = resources + "/visualizations" 
+searches_path = resources + "/searches" 
+
+dashboard_type = "dashboard"
+visualization_type = "visualization"
+search_type = "search"
+
+kibana_index = ".kibana"
+indent_level = 3
+
+def create_document(es_index, es_type, es_id, es_body):
+    return json.dumps(es.create(index=es_index,
+                                doc_type=es_type,
+                                id=es_id,
+                                body=es_body),
+                      indent=indent_level)
+
+def create_document_from_file(es_index, es_type, es_id, path_to_updated_json):
+    content = read_json_from_file(path_to_updated_json)
+    create_document(es_index, es_type, es_id, content)
+
+def get_es_id(filename):
+    return splitext(filename)[0]
+
+def delete_document(es_index, es_type, es_id):
+    es.delete(es_index, es_type, es_id)
+
+def read_json_from_file(file):
+    with open(file) as file_raw:    
+        return json.load(file_raw)
+
+def get_version_of_file(file):
+    return read_json_from_file(file)['version']
+
+def update_existing_document(es_index, es_type, es_id, path_to_updated_json):
+    delete_document(es_index, es_type, es_id)
+    create_document_from_file(es_index, es_type, es_id, path_to_updated_json)
+
+def get_request_as_json(es_index, es_type, es_id):
+    return json.loads(json.dumps(es.get(index=es_index, doc_type=es_type, id=es_id)))
+
+
+def load_assets(es_index, es_type, path_to_files, files):
+    for file in files:
+        logging.debug("--------- " + file + " ---------")
+        full_file_path = path_to_files + "/" + file
+        es_id = get_es_id(file)
+        if (es.exists(index=es_index, doc_type=es_type, id=es_id)):
+            get_response_json = get_request_as_json(es_index, es_type, es_id)
+            version_of_disk_file = get_version_of_file(full_file_path)
+            version_of_es_file = get_response_json['_source']['version']
+            if (version_of_disk_file > version_of_es_file):
```

Comment:
Is version_of_disk guaranteed to be integer? In my experience its going to be a string, you can do type(version_of_disk) to verify. Same comment for version_of_es_file